src/main/java/de/danielbechler/diff/access/Instances.java

```diff
@@ -225,8 +225,7 @@ else if (sourceAccessorType != null)
 				// special handling for beans and arrays should go here
 			}
 		}
-		throw new IllegalArgumentException("Detected instances of different types " + types + ". " +
-				"Instances must either be null or have the exact same type.");
```

Comment:
This is a pretty big change from the original implementation. The exception basically guaranteed, that people don't accidentally compare objects of different types. Now it just silently changes the comparison strategy to "equals only", which is pretty easy to miss. 

I think it would be nice to have a little debug message, to make it slightly more obvious and potentially save some poor souls from countless hours of frustrating debugging. What do you think?